File: WebAPI/Models/Dtos/CreateListItemDto.cs
namespace WebAPI.Models.Dtos
{
    public class CreateListItemDto
    {
        public DateTime DueDate { get; set; }
        public bool IsAllDay { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public RepeatFrequency Frequency { get; set; } = RepeatFrequency.Never;
        public Importance Importance { get; set; } = Importance.Normal;
        public string ListTypeName { get; set; }
        public bool IsNewListType { get; set; } = false;
    }
}

File: WebApp/src/app/models/createListItemDto.ts
import { Importance, RepeatFrequency } from './global/enums';

export class CreateListItemDto {
  dueDate: Date;
  isAllDay: boolean;
  title: string;
  description: string;
  frequency: RepeatFrequency;
  importance: Importance;
  listTypeName: string;
  isNewListType: boolean;

  constructor(obj?: CreateListItemDto) {
    this.dueDate = obj?.dueDate || new Date();
    this.isAllDay = obj?.isAllDay || false;
    this.title = obj?.title || '';
    this.description = obj?.description || '';
    this.frequency = obj?.frequency || RepeatFrequency.Weekly;
    this.importance = obj?.importance || Importance.Normal;
    this.listTypeName = obj?.listTypeName || 'Default';
    this.isNewListType = obj?.isNewListType || false;
  }
}
